code/android-core-library/src/main/java/com/adobe/marketing/mobile/internal/utility/FileUtils.kt

```
@@ -0,0 +1,199 @@
+     * @return true if the contents of the input stream were added to the file,
+     *         false otherwise
+     */
+    @JvmStatic
+    fun readInputStreamIntoFile(file: File?, inputStream: InputStream, append: Boolean): Boolean {
+        return try {
+            FileOutputStream(file, append).use { outputStream ->
+                inputStream.copyTo(outputStream, MAX_BUFFER_SIZE)
+            }
+
+            true
+        } catch (e: Exception) {
+            MobileCore.log(
+                LoggingMode.ERROR,
+                TAG,
+                "Unexpected exception while attempting to write to file: ${file?.path} ($e)"
+            )
+            false
+        }
+    }
+
+    /**
+     * Extracts the zip file to an output directory.
+     *
+     * @param zipFile the zip file that needs to be extracted
+     * @param outputDirectoryPath the destination for the extracted [zipFile] contents
+     * @return true if the zip file has been successfully extracted
+     *         false otherwise
+     */
+    @JvmStatic
+    fun extractFromZip(zipFile: File, outputDirectoryPath: String): Boolean {
+        val folder = File(outputDirectoryPath)
+        if (!folder.exists() && !folder.mkdir()) {
+            MobileCore.log(
+                LoggingMode.WARNING,
+                TAG,
+                "Could not create the output directory $outputDirectoryPath"
+            )
+            return false
+        }
+
+        var entryProcessedSuccessfully = true
+        try {
+            ZipInputStream(FileInputStream(zipFile)).use { zipInputStream ->
+                // get the zipped file list entry
+                var ze = zipInputStream.nextEntry
+                val outputFolderCanonicalPath = folder.canonicalPath
+                if (ze == null) {
+                    // Invalid zip file!
+                    entryProcessedSuccessfully = false
+                    MobileCore.log(LoggingMode.WARNING, TAG, "Zip file was invalid")
```

Nit: Can we fail fast and return `false` here?